Add CSSImportRule.layer document (#24014)
mdn/content: files/en-us/web/api/cssimportrule/layername/index.md

---
title: CSSImportRule.layerName
slug: Web/API/CSSImportRule/layerName
page-type: web-api-instance-property
tags:
  - API
  - CSSOM
  - Property
  - Reference
  - CSSImportRule
  - Cascade layers
  - Read-only
browser-compat: api.CSSImportRule.layerName
---

{{APIRef("CSSOM")}}
The read-only **`layerName`** property of the {{domxref("CSSImportRule")}} interface returns the name of the cascade layercreated by the {{cssxref("@import")}} [at-rule](/en-US/docs/Web/CSS/At-rule).

If the created layer is anonymous, the string is empty (`""`), if no layer has been
created, it is the `null` object.

## Value

A string, that can be empty, or the `null` object.

## Examples

> The following stylesheet includes a single {{cssxref("@import")}} rule.

Reviewer: ```suggestion
The document's single stylesheet contains three {{cssxref("@import")}} rules. The first declaration imports a stylesheet into a named layer. The second declaration imports a stylesheet into an anonymous layer. The third declaration imports a stylesheet without a layer declaration.
```

> Therefore the first item in the list of CSS rules will be a `CSSImportRule`.

Reviewer: ```suggestion
```
the js doesn't include `CSSImportRule`; this made the example more difficult to understand rather than clarifying.

The `layerName` property returns the name of the layer associated with the imported
stylesheet.

```css

> @import url("style1.css") screen layer(layer-1);
> @import url("style2.css") screen layer;
> @import url("style3.css") screen;

Reviewer: ```suggestion
@import url("style1.css") layer(layer-1);
@import url("style2.css") layer;
@import url("style3.css");
```

I think importing the CSS conditionally unnecessarily complicates things.
